### PR TITLE
CommonFunctions with tests

### PR DESCRIPTION
**Pull request by IhateTrains**
I've only included ConvertUTF8ToASCII in EncodingConversions, as that may be enough for Jomini era games.

### CODE REVIEW

File: commonItems/UnitTests/CommonFunctionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace commonItems.UnitTests {
+    public class CommonFunctionsTests {
```
Comment (Idhrendur) [inline commonItems/UnitTests/CommonFunctionsTests.cs:9]: Reorder these to match the order of the function they're testing.

File: commonItems/UnitTests/CommonFunctionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        [Fact]
+        public void LastDigitOneGivesSt() {
```
Comment (Idhrendur) [inline commonItems/UnitTests/CommonFunctionsTests.cs:11]: Find a way to indicate which function each of these is testing. Either make the names fit, or split into multiple test classes.

File: commonItems/UnitTests/CommonFunctionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            Assert.Equal("st", CommonFunctions.CardinalToOrdinal(1));
```
Comment (Idhrendur) [inline commonItems/UnitTests/CommonFunctionsTests.cs:12]: > Static equals overloads are used for comparing instances of two types for reference equality. **This method should not be used for comparison of two instances for equality**. This object will always throw with Assert.Fail. Please use Assert.AreEqual and associated overloads in your unit tests.  

(emphasis mine)

Equal -> AreEqual

Same for all tests. Might have to go back and redo existing tests.

File: commonItems/UnitTests/CommonFunctionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            var input = @"c:\this\is/a/path.txt";
+            Assert.Equal(@"c:\this\is/a/", CommonFunctions.GetPath(input));
+        }
+        [Fact]
+        public void GetPathGetsReversedMixedSlashedPath() {
+            var input = @"/this/is\a\path.txt";
+            Assert.Equal(@"/this/is\a\", CommonFunctions.GetPath(input));
+        }
+        [Fact]
+        public void GetReturnsBlankStringForRawFiles() {
```
Comment (Idhrendur) [inline commonItems/UnitTests/CommonFunctionsTests.cs:85]: GetReturnsBlankStringForRawFiles -> GetPathReturnsBlankStringForRawFiles

File: commonItems/CommonFunctions.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            }
+
+            return tenRemainder switch {
+                1 => "st",
+                2 => "nd",
+                3 => "rd",
+                _ => "th",
+            };
+        }
+        public static string CardinalToRoman(int number) {
```
Comment (Idhrendur) [inline commonItems/CommonFunctions.cs:60]: Add tests

File: commonItems/CommonFunctions.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                number %= num[i];
+                while (div-- > 0) {
+                    sb.Append(sym[i]);
+                }
+                --i;
+            }
+            return sb.ToString();
+        }
+
+        public static string NormalizeStringPath(string stringPath) {
```
Comment (Idhrendur) [inline commonItems/CommonFunctions.cs:76]: Add tests

File: commonItems/CommonFunctions.cs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+            var toReturn = NormalizeUTF8Path(stringPath);
+            toReturn = ReplaceCharacter(toReturn, '-');
+            toReturn = ReplaceCharacter(toReturn, ' ');
+            return toReturn;
+        }
+
+        /// from C++ version's OSCommonLayer
+        public static string NormalizeUTF8Path(string utf8Path) {
```
Comment (Idhrendur) [inline commonItems/CommonFunctions.cs:84]: Add tests

**Review by Idhrendur — APPROVED** [review]

File: commonItems/UnitTests/CommonFunctionsTests.cs (hunk 3 above)
Comment (IhateTrains) [reply commonItems/UnitTests/CommonFunctionsTests.cs:12]: Isn't this from MSUnit documentation? I use the XUnit framework.

File: commonItems/UnitTests/CommonFunctionsTests.cs (hunk 2 above)
Comment (IhateTrains) [reply commonItems/UnitTests/CommonFunctionsTests.cs]: I've split CommonFunctionsTests into multiple test classes.

File: commonItems/UnitTests/CommonFunctionsTests.cs (hunk 1 above)
Comment (IhateTrains) [reply commonItems/UnitTests/CommonFunctionsTests.cs]: This is no longer relevant with the splitting up of CommonFunctionsTests.

File: commonItems/UnitTests/CommonFunctionsTests.cs (hunk 3 above)
Comment (Idhrendur) [reply commonItems/UnitTests/CommonFunctionsTests.cs]: Oh, never mind then.

**Pull request merged**